social-meli/src/test/java/com/api/social_meli/unit/service/UserServiceTest.java

```diff
@@ -1,4 +1,102 @@
 package com.api.social_meli.unit.service;
 
+import com.api.social_meli.dto.GetFollowerCountDto;
+import com.api.social_meli.dto.MessageDto;
+import com.api.social_meli.exception.NotFoundException;
+import com.api.social_meli.model.User;
+import com.api.social_meli.repository.impl.UserRepositoryImpl;
+import com.api.social_meli.service.impl.UserServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
 public class UserServiceTest {
+
+    @Mock
+    private UserRepositoryImpl userRepository;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    //region T-0002 - US-0007 UNFOLLOW USER
+    @Test
+    void unfollowUser_success() {
+        // ARRANGE
+        int userId = 1;
+        int userIdToUnfollow = 2;
+
+        User user = new User();
+        user.setId(userId);
+        user.setFollowed(new ArrayList<>(List.of(userIdToUnfollow)));
```

Comment:
podemos crearnos con el patron factory, en la clase estatica de factory que está en utils, un metodo que nos cree un  y tambien mas cosas si se quieren 
```
User user = UserFactory.createUser(userId, List.of(userIdToUnfollow), new ArrayList<>());
User userToUnfollow = UserFactory.createUser(userIdToUnfollow, new ArrayList<>(), List.of(userId));
```

y tambien se pueden crear en el factory casos de usuarios mas especificos, como UserWithPosts, UserWithoutPosts, etc.

Reply:
y se puede simplificar el new ArrayList<> directamente haciendo
`user.setFollowed(List.of(userIdToUnfollow));`